SurveyTests/Presentation/Modules/Login/LoginViewModelSpec.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  LoginViewModelSpec.swift
+//  SurveyTests
+//
+//  Created by Khanh on 07/08/2022.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+//  swiftlint:disable function_body_length closure_body_length
+
+import Combine
+import Nimble
+import Quick
+import SwiftUI
+
+@testable import Survey
+
+final class LoginViewModelSpec: QuickSpec {
+
+    var logInUseCase: LoginUseCaseMock!
+    var storeTokenUseCase: StoreTokenUseCaseMock!
+    var input: LoginViewModel.Input!
+    var output: LoginViewModel.Output!
+
+    override func spec() {
+        var viewModel: LoginViewModel!
+        let logInTrigger = PassthroughSubject<Void, Never>()
+
+        describe("a LoginViewModel") {
+
+            beforeEach {
+                self.logInUseCase = LoginUseCaseMock()
+                self.storeTokenUseCase = StoreTokenUseCaseMock()
+                viewModel = LoginViewModel(loginUseCase: self.logInUseCase, storeUseCase: self.storeTokenUseCase)
+
+                self.input = LoginViewModel.Input(logInTrigger: logInTrigger.eraseToAnyPublisher())
+                self.output = viewModel.transform(self.input)
+            }
+
+            afterEach {
+                viewModel = nil
```

Comment:
Wonder why you have to set nil to viewModel? Some bug with the test?

Reply:
There's no bug with the test, I set viewModel to nil after every example since i would like to kinda `deinit` viewModel before the next example. But it seems like a redundant code here since that inits viewModel before each every example 🥲 . I will remove it 💯